torch/_functorch/partitioners.py

```diff
@@ -88,6 +88,9 @@ def _extract_graph_with_inputs_outputs(joint_graph, inputs, outputs):
     new_graph = fx.Graph()
     env = {}
 
+    # Ensure we can ask about our input nodes quickly.
+    inputs = inputs if isinstance(inputs, set) else set(inputs)
```

Comment:
since we are iterating over this shortly, and want to have deterministic iteration order, should we instead use `dict.fromkeys` and update whatever is making inputs set in the first place?

Reply:
Lol - I actually meant to put this line after the iteration (and in my internal version I did - I just copied the change over to the OSS version poorly).  But I also realized that it's probably correct to do `node in env` instead.